### ISSUE TITLE
Review code structure and comments

### ISSUE THREAD

**Issue by d13r** [bin-cli/bin-cli#34]
- Consider moving more functionality into separate functions, with a high-level `main()` function to tie it all together
- Consider adding more comments to explain the use and structure of variables and maps
- Consider using [mdsh](https://github.com/bashup/mdsh)

**Comment by d13r** [comment]
I briefly tried formatting the source code as Markdown (for `mdsh`)... While I like the idea, PhpStorm treats each block as a separate file, so can't do things like jumping to declarations. Not surprising really. Plus it means the line numbers don't match up, which makes debugging harder.

**Comment by d13r** [comment]
I [tried mdsh again](https://github.com/bin-cli/bin-cli/tree/mdsh), but ultimately decided the main focus should be on the code not the explanation of the code. Instead, I spent some time reorganising the code, adding more comments, changing variable names, and so on. I decided not to move all code into functions for now, as my first attempt resulted in more fragmented code, but I may reconsider that in the future. I think it is good enough for now.
